sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -466,9 +468,9 @@ class Analyzer(
         // Find aggregate expressions and evaluate them early, since they can't be evaluated in a
         // Sort.
         val (withAggsRemoved, aliasedAggregateList) = newOrdering.map {
-          case aggOrdering if aggOrdering.collect { case a: AggregateExpression => a }.nonEmpty =>
+          case aggOrdering if aggOrdering.find(_.isInstanceOf[AggregateExpression]).isDefined =>
             val aliased = Alias(aggOrdering.child, "_aggOrdering")()
-            (aggOrdering.copy(child = aliased.toAttribute), Some(aliased))
+            (aggOrdering.copy(child = UnresolvedReference(aliased.exprId)), Some(aliased))
```

Comment:
I'm not sure how to solve it by adding a `resolvable` field to `Attribute`, maybe I missed something here?